paymentsheet/src/main/java/com/stripe/android/paymentsheet/wallet/SavedPaymentMethodsSheetResult.kt

```diff
@@ -0,0 +1,45 @@
+package com.stripe.android.paymentsheet.wallet
+
+import androidx.annotation.RestrictTo
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.paymentsheet.model.PaymentOption
+
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+sealed class SavedPaymentMethodsSheetResult {
+    /**
+     * The customer selected a payment method
+     * @param selection, the [PaymentOptionSelection] the customer selected from the
+     * [SavedPaymentMethodsSheet]
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    class Selected internal constructor(
+        val selection: PaymentOptionSelection
+    ) : SavedPaymentMethodsSheetResult()
+
+    /**
+     * The customer canceled the sheet
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    object Canceled : SavedPaymentMethodsSheetResult()
```

Comment:
Worth considering making this just a class? For internal APIs, I'd always use object here. But for public APIs, it's much easier to add a param to a class in the future if we need it. I'm looking at you `PaymentSheetResult.Completed`.